backend/corpora/dataset_processing/remaster_cxg_v2.py

```diff
@@ -0,0 +1,311 @@
+
+    logger.info(f"Downloading {meta_path} to {local_path}/cxg_group_metadata")
+    download_command = ["aws", "s3", "sync", meta_path, f"{local_path}/cxg_group_metadata", "--quiet"]
+    # Let errors fail the pipeline
+    subprocess.run(download_command, check=True)
+    with tiledb.open(f"{local_path}/cxg_group_metadata", "r") as X:
+        if X.meta["cxg_version"] != "0.3.0":
+            evolve_obs(local_path)
+            increment_version(local_path)
+
+            params = {
+                "kind": "auto",
+                "obs_extent": 256,
+                "var_extent": 256,
+                "cell_order": "row",
+                "tile_order": "row",
+                "capacity": 1024000,
+                "compression": 22,
+                "target_array": "X_new",
+                "source_array": "X_old",
+            }
+            try:
+                executed = evolve_X(cxg=local_path, **params)  # executed is true if a sparse array was upgraded
+            except Exception:
+                executed = False
+
+            if executed:
+                logger.info(f"Dataset at {path} computed successfully")
+            else:
+                logger.info("Dataset was dense, X was not upgraded")
+
+            if not dry_run and executed:
+                for suffix in ["r", "c"]:
+                    upload_command = [
+                        "aws",
+                        "s3",
+                        "sync",
+                        "--delete",
+                        f"{local_path}/X_new" + suffix,
+                        path + suffix,
+                        "--quiet",
+                    ]
+                    subprocess.run(upload_command, check=True)
+                    delete_command = ["aws", "s3", "rm", f"{path}", "--recursive", "--quiet"]
+                    subprocess.run(delete_command, check=True)
+            upload_command = ["aws", "s3", "sync", "--delete", f"{local_path}/new_obs", obs_path, "--quiet"]
+            subprocess.run(upload_command, check=True)
+            upload_command = ["aws", "s3", "sync", "--delete", f"{local_path}/cxg_group_metadata", meta_path, "--quiet"]
+            subprocess.run(upload_command, check=True)
+        else:
+            logger.info("Dataset was already upgraded")
+
+    # Cleanup
+    logger.info("Cleaning up local files")
+    _try_to_delete(f"{local_path}/X_old")
+    _try_to_delete(f"{local_path}/X_newr")
+    _try_to_delete(f"{local_path}/X_newc")
+    _try_to_delete(f"{local_path}/old_obs")
+    _try_to_delete(f"{local_path}/new_obs")
+    _try_to_delete(f"{local_path}/cxg_group_metadata")
+
+
+def _try_to_delete(path):
+    import shutil
+
+    try:
+        shutil.rmtree(path)
+    except FileNotFoundError:
+        pass
+
+
+def increment_version(cxg):
+    """
+    Increments the version number of the CXG
+    :param cxg: the path to the CXG
+    """
+    with tiledb.open(f"{cxg}/cxg_group_metadata", "w") as X:
+        X.meta["cxg_version"] = "0.3.0"
```

Comment:
nit: should use the CXG_VERSION constant here rather than hard-coding the version check